.github/workflows/build.yaml

```diff
@@ -40,6 +40,39 @@ jobs:
           BOT: main
         run: ./tool/bots.sh
 
+  dcm:
+    name: Dart Code Metrics
+    needs: flutter-prep
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Clone Flutter DevTools
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
+        with:
+          ref: "${{ github.event.pull_request.head.sha }}"
+      - name: Load Cached Flutter SDK
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84
+        with:
+          path: |
+            ./flutter-sdk
+          key: flutter-sdk-${{ runner.os }}-${{ needs.flutter-prep.outputs.latest_flutter_candidate }}
+      - name: Run tool/bots.sh
+        run: ./tool/bots.sh
```

Comment:
why do we need to run the bots.sh script for running the DCM check?

Reply:
If I remember correctly, it's because that's where we fetch the pub packages. I don't think DCM can analyze the code until the dependencies are fetched.

Reply:
Hmm actually, since this is now in build.yaml, we might be able to remove this step (and the cloning step since that's handled above). Will see if it works removed

Reply:
Ah no, we can't, see https://github.com/flutter/devtools/actions/runs/6724383676/job/18276582319?pr=6635